test/topology_tasks/test_tablet_tasks.py

```diff
@@ -0,0 +1,130 @@
+
+    repair_tasks = await get_user_repair_tasks()
+    while len(repair_tasks) != expected_number:
+        repair_tasks = await get_user_repair_tasks()
+    return repair_tasks
+
+async def check_and_abort_repair_task(tm: TaskManagerClient, servers: list[ServerInfo], module_name: str):
+    async def check_repair_task_status(status):
+        assert status.scope == "table"
+        assert status.kind == "cluster"
+        assert status.type == "user_repair"
+        assert status.keyspace == "test"
+        assert status.table == "test"
+        assert status.is_abortable
+        assert not status.children_ids
+
+    # Wait until user repair task is created.
+    repair_tasks = await wait_for_tasks(tm, servers[0], module_name, 1)
+
+    task = repair_tasks[0]
+    assert task.scope == "table"
+    assert task.keyspace == "test"
+    assert task.table == "test"
+    assert task.state in ["created", "running"]
+
+    status = await tm.get_task_status(servers[0].ip_addr, task.task_id)
+
+    check_repair_task_status(status)
+    assert status.state in ["created", "running"]
+
+    async def wait_for_task():
+        status_wait = await tm.wait_for_task(servers[0].ip_addr, task.task_id)
+        check_repair_task_status(status_wait)
+        assert status_wait.state == "done"
+
+    async def abort_task():
+        await tm.abort_task(servers[0].ip_addr, task.task_id)
+
+    await asyncio.gather(wait_for_task(), abort_task())
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_tablet_repair_task(manager: ManagerClient):
+    module_name = "tablets"
+    tm = TaskManagerClient(manager.api)
+
+    servers, cql, hosts, table_id = await create_table_insert_data_for_repair(manager)
+    assert module_name in await tm.list_modules(servers[0].ip_addr), "tablets module wasn't registered"
+
+    async def repair_task():
+        token = -1
+        # Keep retring tablet repair.
+        await inject_error_on(manager, "repair_tablet_fail_on_rpc_call", servers)
+        await manager.api.tablet_repair(servers[0].ip_addr, "test", "test", token)
+
+    await asyncio.gather(repair_task(), check_and_abort_repair_task(tm, servers, module_name))
+
+async def check_repair_task_list(tm: TaskManagerClient, servers: list[ServerInfo], module_name: str):
+    def get_task_with_id(repair_tasks, task_id):
+        tasks_with_id1 = [task for task in repair_tasks if task.task_id == task_id]
+        assert len(tasks_with_id1) == 1
+        return tasks_with_id1[0]
+
+    # Wait until user repair tasks are created.
+    repair_tasks0 = await wait_for_tasks(tm, servers[0], module_name, len(servers))
+    repair_tasks1 = await wait_for_tasks(tm, servers[1], module_name, len(servers))
+    repair_tasks2 = await wait_for_tasks(tm, servers[2], module_name, len(servers))
```

Comment:
I just want to confirm that all the nodes are supposed to get same repair tasks, no matter where the repair task is created and where the repair task is queried.

Reply:
Ok